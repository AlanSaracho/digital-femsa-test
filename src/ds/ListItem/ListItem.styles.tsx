import React from 'react';
import styled from 'styled-components/native';

export const Container = styled.View`
  flex-direction: row;
  padding-horizontal: 10px;
  padding-vertical: 4px;
  background-color: #FFFFFF;
  align-self: stretch;
`;

export const Avatar = styled.Image`
width: 55px;
height: 55px;
border-radius: 10px;
background-color: #D0D0D0;
margin-right: 11px;
`;

export const MainTextContainer = styled.View`
flex-shrink: 10;

`;

export const Title = styled.Text`
font-family: 'Avenir';
font-weight: 800;
font-size: 14px;
line-height: 19px;
margin-top: 5px;
margin-bottom: 7px;
color: #000000;
`;
export const Description = styled.Text`
font-family: 'Avenir';
font-weight: 400;
font-size: 12px;
line-height: 16px;
color: #000000;
`;

export const Chevron = styled.Image`
align-self: center;
position: absolute;
right: 0;
width: 8px;
height: 8px;
`;
