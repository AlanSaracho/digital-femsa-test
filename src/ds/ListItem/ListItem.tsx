import React from "react"
import { Pressable, Image, ViewStyle } from "react-native"
import { Container, Avatar, MainTextContainer, Title, Description, Chevron } from "./ListItem.styles"

type Props = {
  title: string;
  description: string;
  onPress?: () => void;
  avatarSource: string;
  style?: ViewStyle | ViewStyle[];
  children?: React.ReactNode;
};

export const ListItem: React.FunctionComponent<Props> = ({
   children,
   title,
   description,
   onPress,
   avatarSource,
   style,
  }) => {
  return (
    <Pressable onPress={onPress}>
      <Container style={style}>
        <Avatar source={{ uri: avatarSource }} />
        <MainTextContainer>
          <Title children={title} ellipsizeMode="tail" numberOfLines={1} />
          <Description children={description} />
        </MainTextContainer>
        {children}
      </Container>
    </Pressable>
  )
}