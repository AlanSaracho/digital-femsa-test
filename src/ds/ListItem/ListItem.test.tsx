import React from 'react';
import renderer from 'react-test-renderer';
import { ListItem } from './ListItem';

describe('<ListItem>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ListItem  title='lorem' description='ipsum' avatarSource='lorem'/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
