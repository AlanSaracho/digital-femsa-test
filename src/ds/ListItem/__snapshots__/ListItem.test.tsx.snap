// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#FFFFFF",
          "flexDirection": "row",
          "paddingHorizontal": 10,
          "paddingVertical": 4,
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "lorem",
        }
      }
      style={
        Array [
          Object {
            "backgroundColor": "#D0D0D0",
            "borderRadius": 10,
            "height": 55,
            "marginRight": 11,
            "width": 55,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "flexShrink": 10,
          },
        ]
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Avenir",
              "fontSize": 14,
              "fontWeight": "800",
              "lineHeight": 19,
              "marginBottom": 7,
              "marginTop": 5,
            },
          ]
        }
      >
        lorem
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Avenir",
              "fontSize": 12,
              "fontWeight": "400",
              "lineHeight": 16,
            },
          ]
        }
      >
        ipsum
      </Text>
    </View>
  </View>
</View>
`;
